src/config.js

```diff
@@ -1,31 +1,52 @@
 import { config as envConfig } from 'dotenv';
-import { readFileSync } from 'fs';
 import debug from 'debug';
 
 envConfig();
 
-const { PORT = '1337', HOST = 'localhost' } = process.env;
-
-const packageJSON = JSON.parse(readFileSync('./package.json', 'utf8'));
-
 const DEBUG_PREFIX_APP    = 'app';
 const DEBUG_PREFIX_CONFIG = 'config';
 
-const debugApp    = debug(DEBUG_PREFIX_APP);
 const debugConfig = debug(DEBUG_PREFIX_CONFIG);
 
-//NOTE: You need to tap $env:DEBUG="config" in terminal to turn on debug
-debugApp('%o booting', packageJSON.name);
+class Config {
+    constructor (port = '', host = '') {
+        this._port = port;
+        this._host = host;
+    }
+
+    get port () {
+        const { port = '1337' } = process.env;
+
+        //You need to tap $env:DEBUG="config" in terminal to turn on debug
+        debugConfig(`PORT: ${this._port || port}`);
+
+        return this._port || port;
+    }
 
-debugConfig(`PORT: ${PORT}`);
-debugConfig(`HOST: ${HOST}`);
+    get host () {
+        const { host = 'localhost' } = process.env;
+
+        //You need to tap $env:DEBUG="config" in terminal to turn on debug
+        debugConfig(`HOST: ${this._host || host}`);
+
+        return this._host || host;
+    }
+}
+
+const config = new Config();
+const PORT = config.port;
+const HOST = config.host;
 
 export default {
```

Comment:
Well, from my perspective, there is no point in defining a default export to be just a list of exported objects. You already have a better way to do it with a named export sentence just below in your code.

Also, don't be afraid to alter or even remove tests if they mess with your functionality. Code purity is good, but your program should be functional after all. 

Yet another point: with the current way of exporting `host` and `port` you won't be able to catch configuration changes when you import them directly from the module (`import { host } from './config.js`).

I suggest you to try the following two module approach:
##### globals.js
```js
export let host = 'localhost';
export let port = 1337;

export function update (newValues) {
    if (this)
      return Object.assign(this, newValues);

    host = newValues.host;
    port = newValues.port;

    return { host, port };
}
```

##### config.js
```js
import * as GLOBALS from './GLOBALS';

const DEFAULTS = { ...GLOBALS };

export class Config {
    static get defaults () { return DEFAULTS }
    static get globals () { return GLOBALS }

    constructor (values) {
        this.values = { ...Config.defaults };

        this.values.update(Object.assign(Config._getFromEnv(), values));

        Config.globals.update(this.values);
    }
}

export default Config.globals;

export * from './globals';
```

I haven't tested this in reality, so please try to implement your config this way, and tell if it is possible or not 😉

Reply:
Interesting 😊 I didn't know that we could change imported variables with functions from the same module. But now I think it's logical. Cool 😎